pkg/controller/backup/backup_controller.go

```diff
@@ -186,31 +186,24 @@ func (c *Controller) updateBackup(cur interface{}) {
 		if newBackup.Spec.Mode == v1alpha1.BackupModeLog {
 			return
 		}
-		selector, err := label.NewBackup().Instance(newBackup.GetInstanceName()).BackupJob().Backup(name).Selector()
```

Comment:
Why implement retry on updateBackup, updateBackup should simply do enqueue stuffs? I think the natural place to implement retry logic is on reconcile logic， i.e.  func (bm *backupManager) syncBackupJob(backup *v1alpha1.Backup)

Reply:
yes, it should implement in `syncBackupJob` if it is in the new architecture, but current operator's  old architecture, `syncBackupJob` just check whether job exist or not, it's hard to say it's a reentrant logic. so if we  implement in `syncBackupJob`, i'm afraid it may be more changes than current pr.